### DISCUSSION TITLE
DockPanel-ListBox Visible in Preview but not in Build

### DISCUSSION THREAD

**Discussion by K2kmoments** [AvaloniaUI/Avalonia#15795] [Q&A]
As Described. I am having an issue seeing my "Notebook  list view/Box" content in the app :/ (in preview everything is ok)

<img width="1503" alt="Screenshot 2024-05-22 at 23 08 41" src="https://github.com/AvaloniaUI/Avalonia/assets/29265695/966ebc8b-134f-4b6b-b012-76791db5e2c9">

<Window
    Background="#757575"
    CanResize="False"
    Height="720"
    Title="NotesWindow"
    Width="1200"
    d:DesignHeight="720"
    d:DesignWidth="1200"
    mc:Ignorable="d"
    x:Class="EvernoteClone.Views.NotesWindow"
    x:DataType="vm:NotesViewModel"
    xmlns="https://github.com/avaloniaui"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    xmlns:vm="clr-namespace:EvernoteClone.ViewModels"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml">

    <DockPanel DataContext="{StaticResource ViewModel}" x:DataType="vm:NotesViewModel">
        <Menu DockPanel.Dock="Top">
            <MenuItem Header="File">
                <MenuItem Command="{Binding NewNotebookCommand}" Header="New Notebook" />
                <MenuItem
                    Command="{Binding NewNoteCommand}"
                    CommandParameter="{Binding SelectedNotebook, Mode=TwoWay}"
                    Header="New Note" />
                <Separator />
                <MenuItem Click="MenuItem_OnClick" Header="Exit" />
            </MenuItem>
        </Menu>

        
        <ListBox
            DockPanel.Dock="Left"
            ItemsSource="{Binding Notebooks}"
            SelectedItem="{Binding SelectedNotebook, Mode=TwoWay}"
            Width="140">

            <ListBox.ItemTemplate>
                <DataTemplate>
                    <TextBlock Text="{Binding Name}" VerticalAlignment="Top" />
                </DataTemplate>
            </ListBox.ItemTemplate>
        </ListBox>
        

        <ListBox
            Background="Bisque"
            DockPanel.Dock="Left"
            ItemsSource="{Binding Notes}"
            Width="140">
            <ListBox.ItemTemplate>
                <DataTemplate>
                    <TextBlock Text="{Binding Title}" />
                </DataTemplate>
            </ListBox.ItemTemplate>
        </ListBox>

        <StackPanel
            Background="LightGreen"
            DockPanel.Dock="Top"
            Height="30"
            Orientation="Horizontal">

            <Button Click="SpeechButton_OnClick" Content="Speech" />
            <Button Content="Funct1" />
            <Button Content="Funct2" />
            <Button Content="Funct3" />
            <DropDownButton HorizontalAlignment="Right" />
        </StackPanel>
        <Grid />

    </DockPanel>

</Window>

**Comment by stevemonaco** [comment]
You possibly only have a design-time `DataContext` set and not one at runtime. Or possibly your collection has no items at runtime.

Use DevTools, check the `DataContext` of the `ListBox` at runtime. You can inspect if your `DataContext` has a `Notebooks` collection with items in it. If the `DataContext` is null, then you never set one at runtime.

**Reply by K2kmoments** [reply]
Thank you very much.

I had my Notebook List / Observable object private... 
- I am new in Avalonia AND Rider so I never started the app in debug, By starting in Debug it immediately throw an exception and told me that the class can only access the getter... in "normal Run" the app started without exception
- That the preview showed me the desired outcome was also kind of irritating
- So I guess the Preview ignores private, readonly, etc? 

thanks again :)

greets
